pkg/provider/provider.go

```diff
@@ -156,6 +174,56 @@ func Provider(version string) *schema.Provider {
 }
 
 func providerConfigure(ctx context.Context, d *schema.ResourceData, version string) (interface{}, diag.Diagnostics) {
+	// Check for self-hosted configuration
+	if host := d.Get("host").(string); host != "" {
+		log.Printf("[DEBUG] Configuring self-hosted provider")
+		return configureSelfHosted(ctx, d, version)
+	}
+	log.Printf("[DEBUG] Configuring SaaS provider")
+	return configureSaaS(ctx, d, version)
+}
```

Comment:
This is where we determine if we should use the self-hosted authentication flow or the SaaS authentication flow.